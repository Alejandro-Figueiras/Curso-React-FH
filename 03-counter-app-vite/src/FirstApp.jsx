
const FirstApp = ({ title = 'Hola' }) => {

  return (
    <> {/* <-- Fragment */}
      <h1>{title}</h1>
      <p>Soy un subtitulo</p>
    </>
  )
}

export default FirstApp