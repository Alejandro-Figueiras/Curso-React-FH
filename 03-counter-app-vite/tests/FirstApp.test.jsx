import {render } from '@testing-library/react'
import FirstApp from '../src/FirstApp'

describe("Pruebas en FirstApp", () => {
	// test("debe de hacer match con el snapshot", () => {
  //   const { container } = render(<FirstApp title={"Hola"}/>)
  //   expect(container).toMatchSnapshot();
  // })

  test("debe mostrar el titulo en un h1", () => {
    const { container, getByText, getByTestId } = render(<FirstApp title={"Hola"}/>)
    expect( getByText('Hola') ).toBeTruthy()

    // const h1 = container.querySelector('h1')
    // expect(h1.innerHTML).toContain('Hola')

    expect(getByTestId('test-title').innerHTML).toBe('Hola')
  })
})
