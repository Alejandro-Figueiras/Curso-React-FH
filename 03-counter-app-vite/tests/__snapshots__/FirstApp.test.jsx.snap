// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en FirstApp debe de hacer match con el snapshot 1`] = `
<div>
   
  <h1>
    Hola
  </h1>
  <p>
    Soy un subtitulo
  </p>
</div>
`;
